src/gcovr/__main__.py

```diff
@@ -330,7 +330,7 @@ def main(args=None):
                 LOGGER.debug(f" - {f}")
 
     except re.error as e:
-        LOGGER.error(f"Error setting up filter '{e.pattern}': {e}")
+        LOGGER.error(f"Error setting up filter '{str(e.pattern)}': {e}")
```

Comment:
This str was added because of an issue in mypy. It was something about byte string and string. I'll revisit this.

Reply:
Without the `str` mypy complains:
```
error: If x = b'abc' then f"{x}" or "{}".format(x) produces "b'abc'", not "abc". If this is desired behavior, use f"{x!r}" or "{!r}".format(x). Otherwise, decode the bytes  [str-bytes-safe]
```
I changed to `!s`.

Reply:
Ooh I see, I thought `e` was of type `re.Pattern[str]` but it is actually of type `re.PatternError` (`re.error` until 3.13), for which the type of the `pattern` field cannot be statically known. So the Mypy warning does actually make a lot of sense. In that case, I'd suggest getting rid of the manual quoting:

```diff
-f"Error setting up filter '{str(e.pattern)}': {e}"
+f"Error setting up filter {e.pattern!r}: {e}"
```

Reply:
With `!r` a test is failing:
```
E       assert False
E        +  where False = <built-in method startswith of str object at 0x1033d5c30>("Error setting up filter 'C:\\\\foo\\moo': bad escape \\m at position 7")
E        +    where <built-in method startswith of str object at 0x1033d5c30> = "Error setting up filter 'C:\\\\\\\\foo\\\\moo': bad escape \\m at position 7".startswith
```
According to the error `bad escape \\m...` we should use `!s` instead. What do you think?

Reply:
The `!r` is not *causing* a "bad escape". The test is verifying the presence of this "bad escape" message in the stringified error. But indeed the `!r` changes the escaping in an incompatible way, causing the test to fail:

    expected: "Error setting up filter 'C:\\\\foo\\moo': bad escape \\m at position 7"
    with !r:  "Error setting up filter 'C:\\\\\\\\foo\\\\moo': bad escape \\m at position 7"

Link to the relevant test: https://github.com/gcovr/gcovr/blob/d50d9b8dda2fe2d628950fb66d7f44f5bccea011/tests/test_args.py#L422-L441

This is a user-facing message, so people shouldn't have to understand Python's escaping rules. The `f"... '{e.pattern}': ..."` approach without any escaping is more correct.

I think what a user would actually expect in most cases would be to report the pattern with shell escaping, e.g. as `shlex.quote(e.pattern)`. However, that too will only accept strings.

If we are *really* sure that the argument value will only ever be a string so that the `e.pattern` can only be a string, then the easiest way might be to tell Mypy about our knowledge:

* for example with an `assert isinstance(e.pattern, str)`
* or via a `typing.cast(str, e.pattern)`

Reply:
The same effect is achieved by using `!s` which, in my opinion, is more readable. The pro for the `typing.cast` is that it shows that there is a problem with typing. What is the better approach?

Reply:
Then:

* let's stick with `!s` (which is going to be correct in all known cases)
* maybe add a comment that explains that Mypy things `e.pattern` could be `bytes`
* maybe add a comment that links to this discussion thread

At $work, I frequently use asserts for this kind of stuff, but that's a different codebase with different needs.